web/src/compose_ui.ts

```diff
@@ -187,11 +211,15 @@ export function insert_syntax_and_focus(
     }
 }
 
-export function replace_syntax(old_syntax, new_syntax, $textarea = $("textarea#compose-textarea")) {
+export function replace_syntax(
+    old_syntax: string,
+    new_syntax: string,
+    $textarea = $<HTMLTextAreaElement>("textarea#compose-textarea"),
+): boolean {
     // The following couple lines are needed to later restore the initial
     // logical position of the cursor after the replacement
     const prev_caret = $textarea.caret();
-    const replacement_offset = $textarea.val().indexOf(old_syntax);
+    const replacement_offset = $textarea.val()?.indexOf(old_syntax);
```

Comment:
I think all of these `?`s should be `!`s instead, since this module is expected to always receive a `textarea` that exists in the DOM and thus has a value.